src/System.Management.Automation/engine/remoting/common/RunspaceConnectionInfo.cs

```diff
@@ -2524,102 +2534,86 @@ private static unsafe void FreeArray(byte** arr, int length)
+                    lpStartupInfo.hStdError = new SafeFileHandle(stdErrPipeClient.DangerousGetHandle(), false);
+                    lpStartupInfo.dwFlags = 0x100;
+
+                    // No new window: Inherit the parent process's console window
+                    creationFlags = 0x00000000;
+
+                    // Create the new process in its own group, so that Ctrl+C is not sent to ssh.exe.  We want to handle this
+                    // control signal internally so that it can be passed via PSRP to the remote session.
+                    creationFlags |= CREATE_NEW_PROCESS_GROUP;
+
+                    // Create the new process suspended so we have a chance to get a corresponding Process object in case it terminates quickly.
+                    creationFlags |= CREATE_SUSPENDED;
+
+                    PlatformInvokes.SECURITY_ATTRIBUTES lpProcessAttributes = new PlatformInvokes.SECURITY_ATTRIBUTES();
+                    PlatformInvokes.SECURITY_ATTRIBUTES lpThreadAttributes = new PlatformInvokes.SECURITY_ATTRIBUTES();
+                    bool success = PlatformInvokes.CreateProcess(
+                        null,
+                        cmdLine,
+                        lpProcessAttributes,
+                        lpThreadAttributes,
+                        true,
+                        creationFlags,
+                        IntPtr.Zero,
+                        startInfo.WorkingDirectory,
+                        lpStartupInfo,
+                        lpProcessInformation);
+
+                    if (!success)
+                    {
+                        throw new Win32Exception(Marshal.GetLastWin32Error());
+                    }
 
-                if (stdErrPipeClient != null) { stdErrPipeClient.Dispose(); }
+                    // At this point, we should have a suspended process.  Get the .Net Process object, resume the process, and return.
+                    Process result = Process.GetProcessById(lpProcessInformation.dwProcessId);
+                    uint returnValue = PlatformInvokes.ResumeThread(lpProcessInformation.hThread);
 
-                throw;
-            }
+                    if (returnValue == PlatformInvokes.RESUME_THREAD_FAILED)
+                    {
+                        throw new Win32Exception(Marshal.GetLastWin32Error());
+                    }
 
-            // Create process
-            PlatformInvokes.STARTUPINFO lpStartupInfo = new PlatformInvokes.STARTUPINFO();
-            PlatformInvokes.PROCESS_INFORMATION lpProcessInformation = new PlatformInvokes.PROCESS_INFORMATION();
-            int creationFlags = 0;
+                    stdInPipeServer = null;
+                    stdOutPipeServer = null;
+                    stdErrPipeServer = null;
+                    stdInPipeClient = null;
+                    stdOutPipeClient = null;
+                    stdErrPipeClient = null;
```

Comment:
Why do we assign to null if we will dispose in finally block?

Reply:
We don't own these objects, so we must not dispose if we succeed.

See https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2000#example-1 for an example.